src/lib/Containers/Inbox.tsx

```diff
@@ -31,6 +32,12 @@ export class Inbox extends React.Component<Props, State> {
     this.fetchData = this.fetchData.bind(this)
   }
 
+  componentWillReceiveProps(newProps) {
```

Comment:
So just checking I understand it right: 

Whenever you navigate to the messaging inbox, the property `isVisible` is set in the props, which means the props are updated, and `componentWillReceiveProps` is triggered, upon which any new data can be fetched? And then _just_ the new messages will be fetched and rendered/added as opposed to the whole VC loaded fresh every time (which we do now and is what is generally being fixed here).

Reply:
Yes, that's precisely what happens 😊.

Reply:
🎉